src/herder/HerderSCPDriver.cpp

```diff
@@ -722,4 +728,97 @@ HerderSCPDriver::acceptedCommit(uint64_t slotIndex, SCPBallot const& ballot)
 {
     mSCPMetrics.mAcceptedCommit.Mark();
 }
+
+void
+HerderSCPDriver::setMetricStartTime(uint64_t slotIndex,
+                                    VirtualClock::time_point start,
```

Comment:
I would ditch the `start` parameter as it's always called with `now` (and rename the method to just be `recordSCPEvent`)